src/node.rs

```diff
@@ -462,195 +491,225 @@ impl InMemoryNode {
-            }
-            zksync_types::api::BlockNumber::Pending => {
-                return not_implemented("get_block_by_number__Pending")
-            }
-            zksync_types::api::BlockNumber::Number(ask_number) => {
-                if ask_number != U64::from(reader.current_miniblock) {
-                    let not_implemented_format = format!("get_block_by_number__{}", ask_number);
-                    return not_implemented(&not_implemented_format);
+        let inner = Arc::clone(&self.inner);
+
+        Box::pin(async move {
+            let reader = match inner.read() {
+                Ok(r) => r,
+                Err(_) => return Err(into_jsrpc_error(Web3Error::InternalError)),
+            };
+
+            match block_number {
+                zksync_types::api::BlockNumber::Earliest => {
+                    return Err(into_jsrpc_error(Web3Error::NotImplemented));
+                }
+                zksync_types::api::BlockNumber::Pending => {
+                    return Err(into_jsrpc_error(Web3Error::NotImplemented));
                 }
+                zksync_types::api::BlockNumber::Number(ask_number)
+                    if ask_number != U64::from(reader.current_miniblock) =>
+                {
+                    return Err(into_jsrpc_error(Web3Error::NotImplemented));
+                }
+                _ => {}
             }
-        }
 
-        let txn: Vec<TransactionVariant> = vec![];
-
-        let block = zksync_types::api::Block {
-            transactions: txn,
-            hash: Default::default(),
-            parent_hash: Default::default(),
-            uncles_hash: Default::default(),
-            author: Default::default(),
-            state_root: Default::default(),
-            transactions_root: Default::default(),
-            receipts_root: Default::default(),
-            number: U64::from(reader.current_miniblock),
-            l1_batch_number: Some(U64::from(reader.current_batch)),
-            gas_used: Default::default(),
-            gas_limit: U256::from(ETH_CALL_GAS_LIMIT),
-            base_fee_per_gas: Default::default(),
-            extra_data: Default::default(),
-            logs_bloom: Default::default(),
-            timestamp: Default::default(),
-            l1_batch_timestamp: Default::default(),
-            difficulty: Default::default(),
-            total_difficulty: Default::default(),
-            seal_fields: Default::default(),
-            uncles: Default::default(),
-            size: Default::default(),
-            mix_hash: Default::default(),
-            nonce: Default::default(),
-        };
+            let block = zksync_types::api::Block {
+                transactions: vec![],
+                number: U64::from(reader.current_miniblock),
+                l1_batch_number: Some(U64::from(reader.current_batch)),
+                gas_limit: U256::from(ETH_CALL_GAS_LIMIT),
+                ..Default::default()
```

Comment:
🤯 That's awesome

Reply:
❤️ [Rust book](https://doc.rust-lang.org/book/appendix-03-derivable-traits.html?highlight=..Default%3A%3Adefault()#default-for-default-values)